sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/FileTypeSensorTest.java

```diff
@@ -86,14 +86,21 @@ public void whenProjectOutPathsNotPresent_shouldNotAddProjectInfo_shouldNotLog()
     when(settingsMock.getString("sonar.projectName")).thenReturn("FOO_PROJ");
     when(settingsMock.getString("sonar.projectKey")).thenReturn("FOO_PROJ:GUID");
     when(settingsMock.getString("sonar.projectBaseDir")).thenReturn("BASE DIR");
-    // note that below it's the singular - Path
-    when(settingsMock.getString("sonar.cs.analyzer.projectOutPath")).thenReturn("CS OutPath");
-    when(settingsMock.getString("sonar.vbnet.analyzer.projectOutPath")).thenReturn("VB OutPath");
-    // should query the `getStringArray` method, not the `getString` method like below
-    when(settingsMock.getString("sonar.cs.analyzer.projectOutPaths")).thenReturn("CS OutPaths");
-    when(settingsMock.getString("sonar.vbnet.analyzer.projectOutPaths")).thenReturn("VB OutPaths");
-    addFileToFileSystem("foo.language", Type.MAIN);
-    addFileToFileSystem("bar.language", Type.TEST);
+    // note that below it's the singular in both calls - "projectOutputPath"
+    when(settingsMock.getString("sonar.LANG_KEY.analyzer.projectOutPath")).thenReturn("foo\\.sonarqube\\out\\0");
+    when(settingsMock.getStringArray("sonar.LANG_KEY.analyzer.projectOutPath")).thenReturn(arrayOf("foo\\.sonarqube\\out\\0"));
+
+    sensor.execute(tester);
+
+    assertThat(logTester.logs()).isEmpty();
+    verifyZeroInteractions(projectTypeCollectorMock);
+  }
+
+  @Test
+  public void projectOutPaths_shouldNotBeRetrievedWithGetString() {
+    // the `projectOutPaths` property is multi-value and should be queried via getStringArray()
+    // this test ensures that only `getStringArray` is used, and not `getString`
+    when(settingsMock.getString("sonar.LANG_KEY.analyzer.projectOutPaths")).thenReturn("foo\\.sonarqube\\out\\0");
```

Comment:
If I understand the mechanics of this test, then 
* getStringArray is empty, resulting in empty log
* getString is not empty, if somebody reads that, he'll get nonempty log
Am I missing something? That looks very indirect and unclear.

We should be more straight forward with this test. By not mocking the value and measuring it's consequences, but rather by failing if somebody queries that value instead.

Reply:
I've merged this and the above in one